client/src/components/exercise/lab10/pages/GeneratedData.js

```diff
@@ -0,0 +1,167 @@
+import React, { Fragment, useEffect } from "react";
+import { bindActionCreators } from "redux";
+import { actions as exerciseActions } from "../../../../reducers/lab10/ExerciseReducer";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import { twMerge } from "tailwind-merge";
+import _ from "lodash";
+import AIExplanationCodeBlock from "../components/code/AIExplanationCodeBlock";
+import useScroll from "../../../../use-hooks/useScroll";
+import { EXERCISE_PLAYING } from "../../../../constants/lab10";
+import ExerciseService from "../../../../services/lab10/ExerciseService";
+import { navigate } from "@reach/router";
+
+const GeneratedData = (props) => {
+  useScroll();
+
+  /**
+   * Update lab state onMount
+   */
+  useEffect(() => {
+    props.actions.updateState(EXERCISE_PLAYING);
+  }, []);
+
+  /**
+   * Retrieve user weights, from db, when user data has loaded
+   */
+  useEffect(() => {
+    if (props.user?.userid) {
+      ExerciseService.retrieveWeights(props.user.userid).then((response) => {
+        if (response.ok) {
+          response.json().then((json) => {
+            props.actions.setWeights(json.weights);
+            props.actions.setSessionCounters(json.session);
+          });
+        }
+      });
+    }
+  }, [props.user]);
+
+  /**
+   * Sorts the colors based on their weight in descending order
+   */
+  const keys = Object.keys(props.weights ?? {}).sort((a, b) => {
+    const weightA = props.weights[a],
+      weightB = props.weights[b];
+
+    if (weightA === weightB) {
+      return 0;
+    }
+
+    return weightA < weightB ? 1 : -1;
+  });
+
+  /**
+   * Redirect the user to the following page
+   */
+  const trainAINav = () => {
+    props.actions.incrementTrainingCounter();
+    return navigate("/Lab10/Exercise/SecondTrainingAI");
+  };
+
+  /**
+   * Redirect the user to the following page
+   */
+  const experienceAINav = () => {
+    props.actions.incrementExperienceCounter();
+    return navigate("/Lab10/Exercise/SecondAISimulation");
+  };
+
+  /**
+   * Redirect the user to the following page
+   */
+  const continueNav = () => {
+    return navigate("/Lab10/Reinforcement");
+  };
+
+  return (
+    <div>
+      <Fragment>
+        <div className={"center-div"}>
+          <div className={"guidance margin-bottom-2"}>
+            <p className={"playthrough__sentence tw-text-center tw-mb-0"}>
+              Below is the data you have generated for your AI. As noted
```

Comment:
I think it could possibly be useful to add a small blurb stating that it's impossible to entirely remove all bias but removing as much unwanted bias as possible is best to make the lab feel like it has a bit more of a conclusion

Reply:
Forgot this was mentioned in the meeting